### DISCUSSION TITLE
Add a way to run a script before committing and pushing

### DISCUSSION THREAD

**Discussion by thislooksfun** [JamesIves/github-pages-deploy-action#635] [Ideas]
I am using this action to automatically deploy some autogenerated docs for a project of mine and I want to be able to have multiple versions of those docs at once. I have this mostly solved by using `target-folder`, but I also want to maintain two symlinks to the `latest` and `stable` versions.

Namely I want to be able to do the following:

- [x] Build the documentation into the `/docs` folder.
- [x] Run this action to switch to the `gh-pages` branch and copy `/docs` into the specified `target-dir`.
- [ ] Create/update the `latest` and `stable` symlinks. <-- This is the step that's missing.
- [x] Push the changes.

My naïve solution would be to add an optional `pre-commit` option that is executed as a script before committing.

**Comment by thislooksfun** [comment]
I managed to get the behavior I wanted a different way. https://github.com/thislooksfun/snoots/commit/c29cae58dd5aab335e66c6b1d9a84049b2f08196. I still think this would be useful, but I no longer require it personally.

**Reply by JamesIves** [reply]
This would have been my suggestion. The sole responsibility of this action is to run the deployment job, outside of that it's up to you to run any pre/post scripts as their own steps.
